zerver/lib/actions.py

```diff
@@ -3314,21 +3314,21 @@ def notify_avatar_url_change(user_profile: UserProfile) -> None:
                active_user_ids(user_profile.realm_id))
 
 def do_change_avatar_fields(user_profile: UserProfile, avatar_source: str,
-                            skip_notify: bool=False) -> None:
+                            skip_notify: bool=False, acting_user: Optional[UserProfile]=None) -> None:
     user_profile.avatar_source = avatar_source
     user_profile.avatar_version += 1
     user_profile.save(update_fields=["avatar_source", "avatar_version"])
     event_time = timezone_now()
     RealmAuditLog.objects.create(realm=user_profile.realm, modified_user=user_profile,
                                  event_type=RealmAuditLog.USER_AVATAR_SOURCE_CHANGED,
                                  extra_data={'avatar_source': avatar_source},
-                                 event_time=event_time)
+                                 event_time=event_time, acting_user=acting_user)
```

Comment:
I think you're missing a callpoint in `zproject/backends.py`; that one I'd prefer to have explicitly pass `acting_user=None`.  The goal here is to have code be explicit about how the acting user is, even if it's unknown (as in the case where a synchronization tool is making the change).

Reply:
@timabbott What about the test cases( where these functions are used) should we pass none there explicitly too?